numpy/lib/_format_impl.py

```diff
@@ -952,6 +961,9 @@ def open_memmap(filename, mode='r+', dtype=None, shape=None,
         if dtype.hasobject:
             msg = "Array can't be memory-mapped: Python objects in dtype."
             raise ValueError(msg)
+        # Convert shape to native Python integers (added change)
+        if shape is not None:
+            shape = tuple(np.atleast_1d(shape).tolist())  # Converts to native ints
```

Comment:
This is not a good way to do this for multiple reasons.  Just use something like `tuple(operator.index(length) for length in shape)`.  That still broadens to technically allow non-tuples (we could reject that, but OK), but narrowly converts to integers.

There are some theoretical changes, where `operator.index()` fails but the `repr` is an integer.  But I think we can assume that is too weird to matter (as an exception).